config/v3_0/types/storage.go

```diff
@@ -50,6 +51,18 @@ func (s Storage) Validate() (r report.Report) {
 				r.AddOnError(errors.ErrLinkUsedSymlink)
 			}
 		}
+		for _, d := range s.Directories {
+			if l1.Hard == nil || !*l1.Hard {
+				continue
+			}
+			target := l1.Target
+			if !filepath.IsAbs(target) {
+				target = filepath.Join(l1.Path, l1.Target)
+			}
+			if l1.Target == d.Path {
```

Comment:
Everything before this line can be hoisted outside the loop.

Reply:
Fixed.